lambda/recipes-reindex/src/snapshotRecipeReindex/index.ts

```diff
@@ -0,0 +1,53 @@
+import { PutObjectCommand, S3Client } from '@aws-sdk/client-s3';
+import type { Handler } from 'aws-lambda';
+import { INDEX_JSON } from 'lib/recipes-data/src/lib/constants';
+import type { RecipeIndex } from 'lib/recipes-data/src/lib/models';
+import { recipeIndexSnapshotKey } from '../constants';
+import type {
+	SnapshotRecipeIndexInput,
+	SnapshotRecipeIndexOutput,
+} from '../types';
+import { getConfig } from './config';
+
+const s3Client = new S3Client({ region: process.env['AWS_REGION'] });
+
+export const snapshotRecipeIndexHandler: Handler<
+	SnapshotRecipeIndexInput,
+	SnapshotRecipeIndexOutput
+> = async (state) => {
+	const { ContentUrlBase, RecipeIndexSnapshotBucket } = getConfig();
+	const { executionId } = state;
+
+	const recipeIndexUrl = `https://${ContentUrlBase}/${INDEX_JSON}`;
+	console.log(`Fetching recipe index from ${recipeIndexUrl}`);
+	const recipeIndexSnapshotResponse = await fetch(recipeIndexUrl);
+
+	// eslint-disable-next-line @typescript-eslint/no-unsafe-assignment -- absent Zod types for the index, casting is necessary
+	const recipeIndexSnapshotJson: RecipeIndex =
+		await recipeIndexSnapshotResponse.json();
```

Comment:
Another option is to use `retrieveIndexData` from `lib/recipes-data` which is already typed (retrieves directly from dynamo)

Reply:
I gave this a bash, and it works well but requires a large refactor to ensure we don't pull in irrelevant fail-fast params. I'll ship this as-is, but it'd be nice to come up with a strategy for configuration that ensured there were no surprises when an import unexpectedly implicated environment parameters. I've a few ideas, as I'm sure you do too 😀